src/modules/users/user.controller.ts

```diff
@@ -0,0 +1,37 @@
+import { Controller, Delete, Get, Post, Put } from '@nestjs/common';
+import { UserService } from './user.service';
+// Nhan cac req tu clien -> server va response server -> clienr (data from service)
+@Controller('users')
+export class UserController {
+  constructor(private readonly userService: UserService) {}
+
+  @Get()
+  getUsers(): string {
+    return this.userService.getUsers();
+  }
+
+  @Get('/check-health')
+  checkHealth(): string {
+    return 'Ok';
+  }
+
+  @Post()
+  createUser(): string {
+    return this.userService.createUser();
+  }
+
+  @Get('/:id')
+  detailUser(): string {
+    return this.userService.detailUser();
+  }
+
+  @Put('/:id')
```

Comment:
dùng Patch được không và chúng khác gì nhau

Reply:
put cập nhật toàn bộ tài nguyên yêu cầu đầy đủ details còn patch thì cập nhật 1 phần tài nguyên và linh hoạt cũng như tiết kiệm băng thông